CMakeLists.txt

```diff
@@ -39,8 +32,34 @@ project(
     LANGUAGES
         CXX
 )
-include(CTest)
+
+if(${PROJECT_SOURCE_DIR} STREQUAL ${PROJECT_BINARY_DIR})
+    message(
+        FATAL_ERROR
+        "CMake generation is not allowed within source directory!! Please set a build folder with '-B'!!"
+    )
+endif()
+
 list(PREPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
+
+include(project_options)
+
+set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
```

Comment:
(personal nit): I prefer to avoid `NO/OFF` and `YES/NO` because the `NO` and `ON` are so easily typo'd into each other and flip their meaning.  So I stick to `TRUE` `FALSE` for values.  I realize some var names have better grammar with yes/no or on/off, though.

Reply:
future cleanup ... (I didn't even know true or false was possible looks like CMake 3.13 per CGPT).